src/services/CompareUtils.jl

```diff
@@ -23,11 +24,28 @@ const GeneratedCompareUnion = Union{MeanMaxPPE, VariableNodeData, DFGNodeParams,
                               GenericFunctionNodeData,
                               DFGFactor, DFGFactorSummary, SkeletonDFGFactor}
 
-@generated function ==(x::T, y::T) where T <: GeneratedCompareUnion
-    ignored = [:_internalId]
-    mapreduce(n -> :(x.$n == y.$n), (a,b)->:($a && $b), setdiff(fieldnames(x), ignored))
+function _compVerbose(type, prop, x, y, ignored)
+  @debug "Comparing: $type.$prop: $x ?= $y"
+  # Propagate the ignores if it's a GeneratedCompareUnion
+  if typeof(x) <: GeneratedCompareUnion
+      !(==(x,y, ignored)) && @info("$type.$prop: $x != $y")
+      return ==(x,y, ignored)
+  else
+      # Otherwise ignore ignored
+      !(==(x,y)) && @info("$type.$prop: $x != $y")
+      return ==(x,y)
+  end
+end
```

Comment:
It would still cause #293.
What I have in mind is to have the generated `==` function for performance:
```julia
function ==(x::T, y::T; ignore::Vector{Symbol}=Symbol[]) where T <: GeneratedCompareUnion
  fields = setdiff(fieldnames(T), ignore)
  if @generated
    mapreduce(n -> :(x.$n == y.$n), (a,b)->:($a && $b), fields)
  else
    mapreduce(n -> getfield(x,n) == getfield(y,n), (a,b)-> a && b, fields)
  end
end
```
and then a `==(a,b,verbose; ignored=[])`
where verbose has levels something like:
1: only show type of compared on failure. ie "!= on val"
2: show type of compared and compact print on failure. "!= on value with x=[1,2, ...., 28,29,30], y=x=[1,2, ...., 28,29,30]
3: show everything.

Reply:
Again I suggest NOT using keywords but rather defaults:
`==(a,b,verbose, ignored=[])`

comma not semicolon

Reply:
if we want to look the same as Base.==

Reply:
another suggestion from slack was perhaps:
`==(a,b,verbose::Bool=false;ignore::Vector{Symbol}=[])`

I'd say as long as this does not conflict with Base and verbose + ignore are available.  Just do it and move on.